### ISSUE TITLE
OpenVX Failure  - halfscale_gaussian

### ISSUE THREAD

**Issue by kiritigowda** [GPUOpen-ProfessionalCompute-Libraries/MIVisionX#541]
TOT has a regression failure. Need to find which commit caused the failure. This is  breaking our current CI flow

```
/bin/runvx -affinity:GPU -dump-profile ../MIVisionX/tests/vision_tests/gdfs/29_opticalFlowLK.gdf 
runvx 1.0.0
OK: using AMD OpenVX 1.3.0
WARNING: kernel com.amd.openvx.OpticalFlowPyrLK_XY_XY not supported on GPU -- falling back to CPU
OK: OpenVX using GPU device#0 (gfx906:sramecc-:xnack-) [OpenCL 2.0 ] [SvmCaps 0 0]
csv,HEADER ,STATUS, COUNT,cur-ms,avg-ms,min-ms,clenqueue-ms,clwait-ms,clwrite-ms,clread-ms
Memory access fault by GPU node-1 (Agent handle: 0x557ba374f260) on address 0x7f6b49000000. Reason: Page not present or supervisor privilege.
Aborted (core dumped)
```

**Comment by kiritigowda** [comment]
Test Case - `/bin/runvx -affinity:GPU -dump-profile ../MIVisionX/tests/vision_tests/gdfs/29_opticalFlowLK.gdf `

**Comment by kiritigowda** [comment]
@paveltc can you find which PR broke this?

**Comment by paveltc** [comment]
@kiritigowda I regressed this a month into the past, trying several different commits. Is this even a valid test case? It says it falls back to CPU and then it gives a GPU related error. When I run TOT and specify -affinity:CPU, it runs without issue.
@asalmanp Do you have anything you can add regarding this problem?

**Comment by asalmanp** [comment]
@kiritigowda @paveltc the faulting [gdf](https://github.com/GPUOpen-ProfessionalCompute-Libraries/MIVisionX/blob/master/tests/vision_tests/gdfs/29_opticalFlowLK.gdf) runs multiple nodes as follows (channel_extract->gaussian_pyramid->fast_corners->optical_flow_pyr_lk). Out of these nodes, only the optical_flow_pyr_lk node reverts to CPU and that's where the "falling back to CPU" message is coming from. The rest of the nodes run on GPU and one of them is causing a GPU segfault.

**Comment by kiritigowda** [comment]
Turning off - `MIVisionX/tests/vision_tests/gdfs/29_opticalFlowLK.gdf` test to allow CI test completion.

**Comment by kiritigowda** [comment]
#544 - temporarily enables CI test cycle.

**Comment by kiritigowda** [comment]
Here are two scenarios - @asalmanp @rrawther @hansely 

1 - GDF with just gaussian nodes - Passes

```
OK: using AMD OpenVX 1.3.0
# ago graph dump BEGIN [*INPUT*]
data input_iyuv = image:IYUV,1280,720
data inter_luma = image:U008,1280,720
data inter_new_pyramid = pyramid:U008,1280,720,4,HALF
data inter_old_pyramid = pyramid:U008,1280,720,4,HALF
node org.khronos.openvx.gaussian_pyramid inter_luma inter_new_pyramid attr:AFFINITY:GPU
node org.khronos.openvx.gaussian_pyramid inter_luma inter_old_pyramid attr:AFFINITY:GPU
# ago graph dump END [*INPUT*]
OK: OpenVX using GPU device#0 (gfx906:sramecc-:xnack-) [OpenCL 2.0 ] [SvmCaps 0 0]
# ago graph dump BEGIN [*FINAL*]
data input_iyuv = image:IYUV,1280,720
data inter_luma = image:U008,1280,720
data inter_new_pyramid = pyramid:U008,1280,720,4,HALF
data inter_old_pyramid = pyramid:U008,1280,720,4,HALF
node com.amd.openvx.ChannelCopy_U8_U8 inter_new_pyramid[0] inter_luma attr:AFFINITY:GPU,1
node com.amd.openvx.ChannelCopy_U8_U8 inter_old_pyramid[0] inter_luma attr:AFFINITY:GPU,1
node com.amd.openvx.ScaleGaussianHalf_U8_U8_5x5 inter_new_pyramid[1] inter_new_pyramid[0] attr:AFFINITY:GPU
node com.amd.openvx.ScaleGaussianHalf_U8_U8_5x5 inter_old_pyramid[1] inter_old_pyramid[0] attr:AFFINITY:GPU
node com.amd.openvx.ScaleGaussianHalf_U8_U8_5x5 inter_new_pyramid[2] inter_new_pyramid[1] attr:AFFINITY:GPU
node com.amd.openvx.ScaleGaussianHalf_U8_U8_5x5 inter_old_pyramid[2] inter_old_pyramid[1] attr:AFFINITY:GPU
node com.amd.openvx.ScaleGaussianHalf_U8_U8_5x5 inter_new_pyramid[3] inter_new_pyramid[2] attr:AFFINITY:GPU
node com.amd.openvx.ScaleGaussianHalf_U8_U8_5x5 inter_old_pyramid[3] inter_old_pyramid[2] attr:AFFINITY:GPU
# ago graph dump END [*FINAL*]
csv,HEADER ,STATUS, COUNT,cur-ms,avg-ms,min-ms,clenqueue-ms,clwait-ms,clwrite-ms,clread-ms
csv,OVERALL,  PASS,     1,      ,  0.32,  0.32,  0.02,  0.27,  0.00,  0.00 (median 0.320)
> total elapsed time:   0.00 sec
> graph profile: 
 COUNT,tmp(ms),avg(ms),min(ms),max(ms),DEV,KERNEL
     1,  0.320,  0.320,  0.320,  0.320,GPU,GRAPH
     1,  0.250,  0.250,  0.250,  0.250,GPU,com.amd.openvx.ChannelCopy_U8_U8
     1,  0.279,  0.279,  0.279,  0.279,GPU,com.amd.openvx.ScaleGaussianHalf_U8_U8_5x5
     1,  0.279,  0.279,  0.279,  0.279,GPU,com.amd.openvx.ScaleGaussianHalf_U8_U8_5x5
     1,  0.279,  0.279,  0.279,  0.279,GPU,com.amd.openvx.ScaleGaussianHalf_U8_U8_5x5
     1,  0.280,  0.280,  0.280,  0.280,GPU,com.amd.openvx.ScaleGaussianHalf_U8_U8_5x5
     1,  0.280,  0.280,  0.280,  0.280,GPU,com.amd.openvx.ScaleGaussianHalf_U8_U8_5x5
     1,  0.281,  0.281,  0.281,  0.281,GPU,com.amd.openvx.ScaleGaussianHalf_U8_U8_5x5
OK: OpenCL buffer usage: 3409984, 9/9
```

2 - GDF with channel extract + gaussian nodes - fails - any other combination passes

```
# ago graph dump BEGIN [*INPUT*]
data input_iyuv = image:IYUV,1280,720
data inter_luma = image:U008,1280,720
data inter_new_pyramid = pyramid:U008,1280,720,4,HALF
data inter_old_pyramid = pyramid:U008,1280,720,4,HALF
node org.khronos.openvx.channel_extract input_iyuv scalar:ENUM,CHANNEL_Y inter_luma attr:AFFINITY:GPU
node org.khronos.openvx.gaussian_pyramid inter_luma inter_new_pyramid attr:AFFINITY:GPU
node org.khronos.openvx.gaussian_pyramid inter_luma inter_old_pyramid attr:AFFINITY:GPU
# ago graph dump END [*INPUT*]
OK: OpenVX using GPU device#0 (gfx906:sramecc-:xnack-) [OpenCL 2.0 ] [SvmCaps 0 0]
# ago graph dump BEGIN [*FINAL*]
data input_iyuv = image:IYUV,1280,720
data inter_luma = image:U008,1280,720
data inter_new_pyramid = pyramid:U008,1280,720,4,HALF
data inter_old_pyramid = pyramid:U008,1280,720,4,HALF
node com.amd.openvx.ChannelCopy_U8_U8 inter_luma input_iyuv[0] attr:AFFINITY:GPU,1
node com.amd.openvx.ChannelCopy_U8_U8 inter_new_pyramid[0] inter_luma attr:AFFINITY:GPU,1
node com.amd.openvx.ChannelCopy_U8_U8 inter_old_pyramid[0] inter_luma attr:AFFINITY:GPU,1
node com.amd.openvx.ScaleGaussianHalf_U8_U8_5x5 inter_new_pyramid[1] inter_new_pyramid[0] attr:AFFINITY:GPU
node com.amd.openvx.ScaleGaussianHalf_U8_U8_5x5 inter_old_pyramid[1] inter_old_pyramid[0] attr:AFFINITY:GPU
node com.amd.openvx.ScaleGaussianHalf_U8_U8_5x5 inter_new_pyramid[2] inter_new_pyramid[1] attr:AFFINITY:GPU
node com.amd.openvx.ScaleGaussianHalf_U8_U8_5x5 inter_old_pyramid[2] inter_old_pyramid[1] attr:AFFINITY:GPU
node com.amd.openvx.ScaleGaussianHalf_U8_U8_5x5 inter_new_pyramid[3] inter_new_pyramid[2] attr:AFFINITY:GPU
node com.amd.openvx.ScaleGaussianHalf_U8_U8_5x5 inter_old_pyramid[3] inter_old_pyramid[2] attr:AFFINITY:GPU
# ago graph dump END [*FINAL*]
csv,HEADER ,STATUS, COUNT,cur-ms,avg-ms,min-ms,clenqueue-ms,clwait-ms,clwrite-ms,clread-ms
Memory access fault by GPU node-1 (Agent handle: 0x5562a2c7a410) on address 0x7f9974800000. Reason: Page not present or supervisor privilege.
Aborted (core dumped)
```

**Comment by kiritigowda** [comment]
Here is a single test case to reproduce the issue - `AGO_DUMP_GRAPH=1 ./bin/runvx node org.khronos.openvx.halfscale_gaussian uniform-image:1920,1080,U008,0xaa image:960,540,U008 scalar:INT32,5`

```
AGO_DUMP_GRAPH=1 ./bin/runvx node org.khronos.openvx.halfscale_gaussian uniform-image:1920,1080,U008,0xaa image:960,540,U008 scalar:INT32,5
runvx 1.0.0
OK: using AMD OpenVX 1.3.0
# ago graph dump BEGIN [*INPUT*]
data $1 = image-uniform:U008,1920,1080,170
data $2 = image:U008,960,540
data $3 = scalar:INT32,5
node org.khronos.openvx.halfscale_gaussian $1 $2 $3
# ago graph dump END [*INPUT*]
OK: OpenVX using GPU device#0 (gfx906:sramecc-:xnack-) [OpenCL 2.0 ] [SvmCaps 0 0]
# ago graph dump BEGIN [*FINAL*]
data $1 = image-uniform:U008,1920,1080,170
data $2 = image:U008,960,540
data $3 = scalar:INT32,5
node com.amd.openvx.ScaleGaussianHalf_U8_U8_5x5 $2 $1 attr:AFFINITY:GPU
# ago graph dump END [*FINAL*]
csv,HEADER ,STATUS, COUNT,cur-ms,avg-ms,min-ms,clenqueue-ms,clwait-ms,clwrite-ms,clread-ms
Memory access fault by GPU node-1 (Agent handle: 0x56488c2f5ef0) on address 0x7f6f60600000. Reason: Page not present or supervisor privilege.
Aborted (core dumped)

```

**Comment by kiritigowda** [comment]
@hansely can you look at this issue?

**Comment by rrawther** [comment]
@hansely: Is this test case only failing in OpenCL? In that case why can't we apply the same fix we have for HIP backend to OpenCL as well?

**Comment by kiritigowda** [comment]
Fixed with #707
